File: android/vbot-android-wrapper/src/main/kotlin/com/vpmedia/vbot/wrapper/VBotFirebaseService.kt
package com.vpmedia.vbot.wrapper

import android.content.Context
import android.util.Log
import androidx.annotation.Keep
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

@Keep
class VBotFirebaseService : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        super.onNewToken(token)
        Log.d("VBotFirebaseService", "onNewToken: $token")
        VBotAndroidWrapper.getInstance(applicationContext).setFirebaseToken(token)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        val data = remoteMessage.data
        Log.d("VBotFirebaseService", "onMessageReceived: $data")
        if (data.containsKey("transId")) {
            VBotAndroidWrapper.getInstance(applicationContext).handleIncomingPush(data)
        }
    }
}

File: VBotMauiApp/Platforms/Android/VBotFirebaseService.cs
using System;
using System.Collections.Generic;
using Android.App;
using Android.Content;
using Android.OS;
using Firebase.Messaging;
using VBotMauiApp.Services;

namespace VBotMauiApp.Platforms.Android;

/// <summary>
/// Service tiếp nhận FCM Push Notification khi có cuộc gọi đến
/// </summary>
[Service(Exported = false)]
[IntentFilter(new[] { "com.google.firebase.MESSAGING_EVENT" })]
public class VBotFirebaseService : FirebaseMessagingService
{
    public override void OnNewToken(string token)
    {
        base.OnNewToken(token);
    }

    public override void OnMessageReceived(RemoteMessage message)
    {
        base.OnMessageReceived(message);
        var data = message.Data;

        if (data != null && data.ContainsKey("transId"))
        {
            var map = new Dictionary<string, string>();
            foreach (var kv in data)
            {
                map[kv.Key] = kv.Value;
            }

            VBotPhoneService.HandleIncomingPushNotification(map);
        }
    }
}


